src/EditorFeatures/CSharpTest/ExtractMethod/ExtractMethodTests.cs

```diff
@@ -7373,28 +7373,7 @@ static void Main(string[] args)
         }
     }
 }";
-            var expected = @"class Test
-{
-    void method()
-    {
-        static void Main(string[] args)
-        {
-            int v = 0;
-            for(int i=0 ; i<5; i++)
-            {
-                v = NewMethod(v, i);
```

Comment:
> To elaborate, my point is that you cannot expect flow analysis to give correct results in this situation.

Flow analysis is an implementation detail.  All i'm saying is that i don't want this scenario to degrade.

> If you want extract method to work anyway, that's a different story

Well... it's not a different story.  As part of your change, you're making it so htat ExtractMethod doesn't work here.  That's the main point i'm addressing.  :)   This is a regression in behavior, and it's in a case where i would not want things to regress.

These tests are not 'flow analysis tests'.  They're specifically capturing the behavior of these features.  And this behavior change seems onerous and undesirable.   That's all :)